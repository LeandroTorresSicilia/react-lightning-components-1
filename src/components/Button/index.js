/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default class Button extends Component {
    getBtnClass() {
        const { variant, className } = this.props;

        return classnames('slds-button', {
            'slds-button--neutral': variant === 'neutral',
            'slds-button--brand': variant === 'brand',
            'slds-button--inverse': variant === 'inverse',
            'slds-button--destructive': variant === 'destructive',
            'slds-button--success': variant === 'success',
        }, className);
    }

    render() {
        const {
            label,
            disabled,
            tabIndex,
            onFocus,
            onBlur,
            onClick,
            style,
            children,
            title,
            ariaHaspopup,
        } = this.props;

        return (
            <button className={this.getBtnClass()}
                    disabled={disabled}
                    tabIndex={tabIndex}
                    onFocus={onFocus}
                    onBlur={onBlur}
                    onClick={onClick}
                    title={title}
                    style={style}
                    aria-haspopup={ariaHaspopup} >
                { !children ? label : children }
            </button>
        );
    }
}

Button.propTypes = {
    /** Label for the button. If a children is specified, it will be displayed.
     Otherwise, the component will expect a label which will then displayed */
    label: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node,
    ]),
    /** Indicates that the element has a popup context menu or sub-level menu. It is
     used for srceen readers */
    ariaHaspopup: PropTypes.bool,
    /** A button predefined style. It must be a string */
    variant: PropTypes.oneOf(['neutral', 'brand', 'inverse', 'destructive', 'success', 'bare']),
    /** This is what will be displayed inside the button. It has precedence over
     the label prop */
    children: PropTypes.node,
    /** Class for custom styles */
    className: PropTypes.string,
    /** Disables the button if set to true */
    disabled: PropTypes.bool,
    /** Tab index */
    tabIndex: PropTypes.number,
    /** Callback function fired when the button is clicked */
    onClick: PropTypes.func,
    /** Callback function fired when the button is focused */
    onFocus: PropTypes.func,
    /** Callback function fired when the button is blurred */
    onBlur: PropTypes.func,
    /** Object with the custom styles. The properties must be
     in camelCase naming convention (e.g. { fontFamily: ‘helvetica’ }) */
    style: PropTypes.object,
    /** This is a description that is showed when a user hover the button */
    title: PropTypes.string,
};

Button.defaultProps = {
    ariaHaspopup: false,
    variant: 'neutral',
    disabled: false,
    onClick: () => {},
    onFocus: () => {},
    onBlur: () => {},
};
