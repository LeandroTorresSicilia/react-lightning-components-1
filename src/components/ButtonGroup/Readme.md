ButtonGroup expamples:

    <div>
        <div className="slds-p-bottom--medium">
            <ButtonGroup>
                <Button label="Refresh" />
                <Button label="Edit" />
                <Button label="Save" />
            </ButtonGroup> 
        </div>
        <div className="slds-p-bottom--medium">
            <ButtonGroup variant="list">
                <Button label="Refresh" variant="brand" />
                <Button label="Edit" variant="brand" />
                <Button label="Save" variant="brand" />
            </ButtonGroup> 
        </div>
    </div>