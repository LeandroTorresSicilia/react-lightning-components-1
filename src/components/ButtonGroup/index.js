/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default class ButtonGroup extends Component {
    getContainerClass() {
        const { className, variant } = this.props;

        return classnames({
            'slds-button-group': variant === 'group',
            'slds-button-group-list': variant === 'list',
        }, className);
    }

    render() {
        const { children, style } = this.props;

        return (
            <div className={this.getContainerClass()} style={style} role="group">
                { children }
            </div>
        );
    }
}

ButtonGroup.propTypes = {
    /** Class for custom styles */
    className: PropTypes.string,
    /** Object with the custom styles. The properties must be in camelCase naming
     convention (e.g. { backgroundColor: green }) */
    style: PropTypes.object,
    /** The list of buttons component or button element */
    children: PropTypes.node,
    /** It is the button group variant */
    variant: PropTypes.oneOf(['group', 'list']),
};

ButtonGroup.defaultProps = {
    variant: 'group',
};
