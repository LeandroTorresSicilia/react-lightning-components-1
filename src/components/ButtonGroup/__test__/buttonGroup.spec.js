import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ButtonGroup from './../index';
import Button from './../../Button';

describe('<ButtonGroup/>', () => {
    it('should render the right markup with the right classes', () => {
        const component = renderer.create(
            <ButtonGroup>
                <Button label="Refresh" />
                <Button label="Edit" />
                <Button label="Save" />
            </ButtonGroup>,
        );

        expect(component).toMatchSnapshot();
    });

    it('should be accessible, the div element should have role="group"', () => {
        const component = shallow(
            <ButtonGroup>
                <Button label="Refresh" />
                <Button label="Edit" />
                <Button label="Save" />
            </ButtonGroup>,
        );

        expect(component.find('div[role="group"]').exists()).toBe(true);
    });
});
