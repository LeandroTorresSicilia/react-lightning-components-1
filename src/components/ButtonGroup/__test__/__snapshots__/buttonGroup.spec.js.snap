// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup/> should render the right markup with the right classes 1`] = `
<div
  className="slds-button-group"
  role="group"
  style={undefined}
>
  <button
    aria-haspopup={false}
    className="slds-button slds-button--neutral"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    style={undefined}
    tabIndex={undefined}
    title={undefined}
  >
    Refresh
  </button>
  <button
    aria-haspopup={false}
    className="slds-button slds-button--neutral"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    style={undefined}
    tabIndex={undefined}
    title={undefined}
  >
    Edit
  </button>
  <button
    aria-haspopup={false}
    className="slds-button slds-button--neutral"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    style={undefined}
    tabIndex={undefined}
    title={undefined}
  >
    Save
  </button>
</div>
`;
